### DISCUSSION TITLE
How to configure 720P@120fps on wfb_gs on ubuntu(not OpenIPC gs)

### DISCUSSION THREAD

**Discussion by lida2003** [svpcom/wfb-ng#336] [Questions] [answered]
As I have recently got SCC338Q+IMX415, 3536DV100 with pre-installed OpenIPC.

It's quite easy to configure 720P@60FPS on airunit(/etc/majestic.yaml) and gs(/etc/vdec.conf)

But I didn't find /etc/vdec.conf on ubuntu custom build version?

So how can I configure SCC338Q+IMX415(720P@120FPS) + ubuntu(?????)?

**Comment by svpcom** [comment]
Just use standard gstreamer pipeline or wfb-ng-osd

**Reply by lida2003** [reply]
Well, it's very strange that when I configure to use H265 codec. The screen is blurred. 
Both 3536 and wfb-ng-osd is blurred, anything to do with my configuration?

BTW, only change codec to H264, then everything is fine.

- gs(/etc/vdec.conf) 1024x768@60FPS h265
```
root@openipc-hi3536dv100:~# cat /etc/vdec.conf
### Port for video rx (default: 5000)

port=5600

### Codec: h264 or h265 (default: h264)

codec=h265

### Incoming data format: stream or frame (default: stream)

format=frame

### Screen output mode:
# 720p60       - 1280 x 720  @ 60 fps
# 1080p60      - 1920 x 1080 @ 60 fps
# 1024x768x60  - 1024 x 768  @ 60 fps
# 1366x768x60  - 1366 x 768  @ 60 fps
# 1280x1024x60 - 1280 x 1024 @ 60 fps
# 1600x1200x60 - 1600 x 1200 @ 60 fps
# 2560x1440x30 - 2560 x 1440 @ 30 fps

mode=1024x768x60

### Mavlink port for telemetry (default: 14750)

mavlink_port=14750

### Osd: none, simple or custom for use own lvgl based osd (custom osd binary need to be upload to /usr/bin directory) (default: simple)

osd=simple

### Records archive (Mount ssd, set up full path to records archive, file names generated automatically. example: /mnt/sda1/recorder)

records=

### Extra param (write in one line split by space):

#    --ar [mode]        - Aspect ratio mode               (Default: keep)
#      keep             - Keep stream aspect ratio
#      stretch          - Stretch to output resolution
#      manual           - Manual image size definition

#    --ar-bg-r [Value]  - Fill color red component      (Default: 0)
#    --ar-bg-g [Value]  - Fill color green component    (Default: 0)
#    --ar-bg-b [Value]  - Fill color blue component     (Default: 0)

#    --ar-x [Value]     - Image position X
#    --ar-y [Value]     - Image position Y
#    --ar-w [Value]     - Image width
#    --ar-h [Value]     - Image height

#    --bg-r [Value]     - Background color red      (Default: 0)
#    --bg-g [Value]     - Background color green    (Default: 96)
#    --bg-b [Value]     - Background color blue     (Default: 0)

extra="--bg-r 30 --bg-g 0 --bg-b 30"
```


- airunit(/etc/majestic.yaml) 1024x768@60FPS h265
```
root@openipc-ssc338q:~# cat /etc/majestic.yaml
system:
  staticDir: /var/www
  webPort: 80
  httpsPort: 443
  plugins: false
  buffer: 1024
isp:
  antiFlicker: disabled
  exposure: 20
  aGain: 40
  sensorConfig: /etc/sensors/imx415_fpv.bin
image:
  mirror: false
  flip: false
  rotate: 0
  contrast: 50
  hue: 50
  saturation: 50
  luminance: 50
video0:
  enabled: true
  codec: h265
  fps: 60
  bitrate: 2048
  rcMode: cbr
  gopSize: 1
  size: 1024x768
video1:
  enabled: false
  codec: h264
  size: 704x576
  fps: 15
jpeg:
  enabled: false
  qfactor: 50
  fps: 5
osd:
  enabled: false
  font: /usr/share/fonts/truetype/UbuntuMono-Regular.ttf
  template: "%d.%m.%Y %H:%M:%S"
  posX: 16
  posY: 16
audio:
  enabled: false
  volume: 30
  srate: 8000
  codec: opus
  outputEnabled: false
  outputVolume: 30
rtsp:
  enabled: true
  port: 554
nightMode:
  enabled: false
  irCutSingleInvert: false
  colorToGray: true
  irSensorPinInvert: false
motionDetect:
  enabled: false
  visualize: false
  debug: false
records:
  enabled: false
  path: /mnt/mmcblk0p1/%F/%H.mp4
  maxUsage: 95
outgoing:
  enabled: true
  server: udp://127.0.0.1:5600
watchdog:
  enabled: true
  timeout: 300
hls:
  enabled: false
onvif:
  enabled: false
ipeye:
  enabled: false
youtube:
  enabled: false
```

![5cc22afb55005e0d0707d68fd57a9a5](https://github.com/svpcom/wfb-ng/assets/1296171/2d929f5a-17bc-4678-b0fd-593a7da01458)

**Reply by lida2003** [reply]
OK, it works. But still doesn't have any idea about [outgoing.naluSize ](https://github.com/OpenIPC/wiki/blob/ca6580572f3ef228a6ca1ffa7ae1efcb8d6e0a18/en/majestic-config.md?plain=1#L114). Anyone please light me up, thanks.

```
root@openipc-ssc338q:~# cli -s .outgoing.naluSize 1200
```

![图片](https://github.com/svpcom/wfb-ng/assets/1296171/0643b770-c25d-4795-93a8-821fa9ac4762)

**Comment by lida2003** [comment]
OK, now met new issue, compiling (`make codec=h265`) wfb_ng_osd in jetson orin, it seems that osd still uses h264 codec.

![图片](https://github.com/svpcom/wfb-ng/assets/1296171/6df8e18c-fd64-453c-8d43-2cd1f266f7b9)

--- EDIT, try "-5"option, still get stucked. It's ok to use gstreamer in Jetson to launch the video @ rtp 5600.

![图片](https://github.com/svpcom/wfb-ng/assets/1296171/22d69ebc-cef4-4793-8539-63c9c57f7340)

**Comment by lida2003** [comment]
> OK, now met new issue, compiling (`make codec=h265`) wfb_ng_osd in jetson orin, it seems that osd still uses h264 codec.
> 
> ![图片](https://private-user-images.githubusercontent.com/1296171/324245020-6df8e18c-fd64-453c-8d43-2cd1f266f7b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3MzYyOTQsIm5iZiI6MTcxMzczNTk5NCwicGF0aCI6Ii8xMjk2MTcxLzMyNDI0NTAyMC02ZGY4ZTE4Yy1mZDY0LTQ1M2MtOGQ0My0yY2QxZjI2NmY3YjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjFUMjE0NjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDgzZjI2M2JmZGM3MGRjZTJmNTA0Mjk4YmQzNTI2NzYzMmFmMzI0NGQxZDNhN2Q2MTJlZjEzNjIzZDg0NzI4MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.e9-0GROfuCj6qNILdquycHDwpg10F0HRGdjHixcWOWU)
> 
> --- EDIT, try "-5"option, still get stucked. It's ok to use gstreamer in Jetson to launch the video @ rtp 5600.
> 
> ![图片](https://private-user-images.githubusercontent.com/1296171/324245318-22d69ebc-cef4-4793-8539-63c9c57f7340.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3MzYyOTQsIm5iZiI6MTcxMzczNTk5NCwicGF0aCI6Ii8xMjk2MTcxLzMyNDI0NTMxOC0yMmQ2OWViYy1jZWY0LTQ3OTMtODUzOS02M2M5YzU3ZjczNDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjFUMjE0NjM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjZiY2Y4MDY4NGYxNDAzZmE0ODI1NmYyZGMxMmE5NjUxYjE2N2I4NTgxNzc1MjUwMGVhOWE0ZThiOGU0MGE2NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.hHeERfLy2hhwvJl14mrCia3_BHIDtfxd06sij4pDHLE)

@svpcom OK, as you have mentioned in https://github.com/svpcom/wfb-ng/discussions/337#discussioncomment-9181669

Anyway, it can be decoded by [gstDecoder](https://github.com/dusty-nv/jetson-utils/blob/af7c667c6367fe6599d8046c213d0cb3b8221ae3/video/videoSource.cpp), which is jetson orin demo code.

Is it possible to port this code to wfb-ng-osd?

**Reply by svpcom** [reply]
Stream produced by majestic (NOT venc) should be decoded by wfb-ng-osd. Try wfb-ng-osd from master branch

**Reply by lida2003** [reply]
Currently, it's latest master, I think the video stream is not from venc, it's from wfb_tx ??? am I right?

- 748 root     wfb_tx -p 0 -u 5600 -R 456000 -K /etc/drone.key -B 20 -M 1 -S 0 -L 0 -G long -k 8 -n 12 -T 0 -i 7669206 -f data wlan0

```
$ git log -n 1
commit 1acd315d187f399199945c3431b5789dd9bd7732 (HEAD -> master, origin/master, origin/HEAD)
Author: Vasily Evseenko <svpcom@gmail.com>
Date:   Fri Dec 15 19:41:16 2023 +0300

    Fix video format in test example
```

IMX415 ==> SCC338Q ==> WiFi ==>wfb_ng(Jetson Orin) ==>wfb_ng_osd(Jetson Orin)

**Reply by svpcom** [reply, marked as answer]
wfb_rx/tx is just a transport for udp packets.  openipc camera can produce RTP-compatible video streams using majectic streamer or **proprietary** format using venc streamer. RTP can be decode by gstreamer or any other standard tool, but proprietary format can be only decoded by **vdec**

**Reply by lida2003** [reply]
> wfb_rx/tx is just a transport for udp packets. 

OK.

> openipc camera can produce RTP-compatible video streams using majectic streamer or **proprietary** format using venc streamer.

Maybe I have found something of enc process. I'll take a look at OpenIPC now.

- 772 root     [venc0_P0_MAIN]
- 773 root     [venc1_P0_MAIN]


Thanks for the help!
